src/Factory/QueryCacheFactory.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Factory;
+
+use Psr\Container\ContainerInterface;
+use RuntimeException;
+use Yiisoft\Db\Cache\QueryCache;
+use Yiisoft\Factory\Definitions\DefinitionInterface;
+use Yiisoft\Factory\Definitions\Normalizer;
+use Yiisoft\Factory\Factory;
+
+final class QueryCacheFactory extends Factory
+{
+    private static ?ContainerInterface $container = null;
+    private static ?DefinitionInterface $definition = null;
+    private static Factory $queryCacheFactory;
+
+    private function __construct(ContainerInterface $container = null)
+    {
+        self::$container = $container;
+        self::$definition = Normalizer::normalize(QueryCache::class);
+    }
+
+    public static function initialize(ContainerInterface $container = null): void
+    {
+        self::$queryCacheFactory = new self($container);
+    }
+
+    /**
+     * Creates a `QueryCache` instance.
+     *
+     * @throws RuntimeException If the created object is not an instance of the `QueryCache`.
+     *
+     * @return QueryCache The `QueryCache` instance.
+     *
+     * @psalm-suppress RedundantConditionGivenDocblockType
+     * @psalm-suppress DocblockTypeContradiction
+     */
+    public static function run(): QueryCache
```

Comment:
```suggestion
    public static function create(): QueryCache
```

Reply:
Can remove inheritance, as it is redundant here. Then you can rename the method from `run()` to `create()`.

Reply:
Done.